notion_client/client.py

```diff
@@ -61,8 +58,17 @@ def _build_request(self, method, path, body):
         return self.client.build_request(method, path, json=body)
 
     def request(self, path, method, query=None, body=None, auth=None):
-        request = self._build_request(method, path, body)
-        return self.client.send(request)
+        try:
+            request = self._build_request(method, path, body)
+            response = self.client.send(request)
+            response.raise_for_status()
+            return response
+        except Exception as e:
+            request_error = build_request_error(e)
+            if request_error is None:
+                raise e
+
+            raise request_error
```

Comment:
Would it be easier to read with something like this?

```suggestion
        except Exception as error:
            raise build_request_error(error) or error
```

Reply:
sounds great

Reply:
changed on latest commits